pkg/apis/edge/v1alpha1/edge-placement.go

```diff
@@ -102,6 +116,35 @@ type NonNamespacedObjectReferenceSet struct {
 	LabelSelectors []metav1.LabelSelector `json:"labelSelectors,omitempty"`
 }
 
+// UpsyncSet specifies a set of objects,
+// which may be namespaced or cluster-scoped,
+// from one particular API group.
+// An object is in this set if:
+// - its API group is the one listed;
+// - its resource (lowercase plural form of object type) is one of those listed;
+// - EITHER the resource is cluster-scoped OR the object's namespace matches `namespaces`; and
+// - the object's name matches `names`.
+type UpsyncSet struct {
+	// `apiGroup` is the API group of the referenced object, empty string for the core API group.
+	APIGroup string `json:"apiGroup,omitempty"`
+
+	// `resources` is a list of lowercase plural names for the sorts of objects to match.
+	// An entry of `"*"` means that all match.
+	// Empty list means nothing matches.
+	Resources []string `json:"resources"`
+
+	// `namespaces` is a list of acceptable namespaces.
+	// An entry of `"*"` means that all match.
+	// Empty list means nothing matches (you probably do not want this
+	// for namespaced resources).
+	Namespaces []string `json:"namespaces,omitempty"`
+
+	// `Names` is a list of objects that match by name.
+	// An entry of `"*"` means that all match.
+	// Empty list means nothing matches (you probably never want an empty list).
+	Names []string `json:"names,omitempty"`
+}
```

Comment:
I think we will need additional criteria to select objects to upsync, such as label selector, and possibly a way to express selection of objects based on presence of ownerReferences to downsynced object.

Reply:
It is easy enough to imagine such a thing.  I would like to hear whether the C2P work needs it.

Reply:
@yuji-watanabe-jp ?  @yana1205 ?

Reply:
good to have flexible selector (namespace selector and resources selector)
- namespace ... label
- resource ... GVK, label, annotation + namespace selector

I agree that selector based on ownerReferences might be useful, but raises the implementation concern of syncer. Chasing owner to the downsynced object might not be always possible since syncer's access is constrained only to downsynced/upsynced objects with RBAC, but ownership link is not always directly among them. So, maybe, we can only support direct-ownership only.

Reply:
In the C2P work, what would go wrong if the objects to upsync could be identified  _only_ by API group and resource (equivalently, Kind)?

Reply:
not sure if the current selector is enough for supporting all the PEP/PVPs at this point, but no issue for demonstrating Kyverno scenario. Let's revisit this later when we have edge-syncer and conduct more expanded and complicated integration. I believe this flexibility is the key if we use this as a configuration management platform for generic scenarion rather than just for kyverno.

Reply:
@MikeSpreitzer I suppose that would be ok for C2P. There are other scenarios (like kcp + Crossplane) where the objects to upsync are secrets. But we may not want to upsync all secrets for security reasons. In any case, I am fine to leave this for later.